connector/connect/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -65,11 +65,25 @@ message SQL {
 message Read {
   oneof read_type {
     NamedTable named_table = 1;
+    FileDataSource file_data_source = 2;
   }
 
   message NamedTable {
     repeated string parts = 1;
   }
+
+  message FileDataSource {
+    repeated string path = 1;
+    string format = 2;
+    string schema = 3;
+    repeated DataSourceOptionPair options = 4;
+  }
+
+  message DataSourceOptionPair {
+    // Option key is case insensitive.
+    string option_key = 1;
+    string option_value = 2;
```

Comment:
Sorry for the late reply, isn't the option_value actually more a literal?

Reply:
They are strings: https://github.com/apache/spark/blob/357193d55508051369228c5e7e252c17089da317/sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala#L703